src/shared/tooltip.rs

```diff
@@ -41,22 +55,72 @@ live_design! {
 
 #[derive(Live, LiveHook, Widget)]
 pub struct Tooltip {
-    #[deref]
-    view: View,
+    #[rust]
+    opened: bool,
+
+    #[live]
+    #[find]
+    content: View,
+
+    #[redraw]
+    #[rust(DrawList2d::new(cx))]
+    draw_list: DrawList2d,
+
+    #[live]
+    draw_bg: DrawQuad,
+    #[layout]
+    layout: Layout,
+    #[walk]
+    walk: Walk,
 }
 
 impl Widget for Tooltip {
     fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
-        self.view.handle_event(cx, event, scope);
+        self.content.handle_event(cx, event, scope);
     }
 
-    fn draw_walk(&mut self, cx: &mut Cx2d, scope: &mut Scope, walk: Walk) -> DrawStep {
-        self.view.draw_walk(cx, scope, walk)
+    fn draw_walk(&mut self, cx: &mut Cx2d, scope: &mut Scope, _walk: Walk) -> DrawStep {
+        self.draw_list.begin_overlay_reuse(cx);
+
+        cx.begin_pass_sized_turtle(self.layout);
+        self.draw_bg.begin(cx, self.walk, self.layout);
+
+        if self.opened {
+            let _ = self.content.draw_all(cx, scope);
+        }
+
+        self.draw_bg.end(cx);
+
+        cx.end_pass_sized_turtle();
+        self.draw_list.end(cx);
+
+        DrawStep::done()
     }
 }
 
 impl TooltipRef {
-    pub fn set_text(&mut self, s:&str){
-        self.label(id!(label)).set_text(s);
+    pub fn set_text_and_pos(&mut self, cx: &mut Cx, text: &str, pos: DVec2) {
```

Comment:
I went further and only leave two functions: `show(cx, position, text)` and `hide()`.

This is because I'm pretty sure you will want to provide a text and position when displaying a tooltip, so let's do it in one shot.